marc.py

```diff
@@ -0,0 +1,612 @@
+        genres = []
+        if isinstance(work, BaseMaterializedWork):
+            # TODO: This is inefficient.
+            genres = work.license_pool.work.genres
+        else:
+            genres = work.genres
+
+        for genre in genres:
+            record.add_field(
+                Field(
+                    tag="650",
+                    indicators=["0", "7"],
+                    subfields=[
+                        "a", genre.name,
+                        "2", "Library Simplified",
+                    ]))
+
+    @classmethod
+    def add_ebooks_subject(cls, record):
+        # This is a general subject that can be added to all records.
+        record.add_field(
+            Field(
+                tag="655",
+                indicators=[" ", "0"],
+                subfields=[
+                    "a", "Electronic books.",
+                ]))
+
+
+class MARCExporter(object):
+    """Turn a work into a record for a MARC file."""
+
+    NAME = ExternalIntegration.MARC_EXPORT
+
+    DESCRIPTION = _("Export metadata into MARC files that can be imported into an ILS manually.")
+
+    # MARC organization codes are assigned by the
+    # Library of Congress and can be found here:
+    # http://www.loc.gov/marc/organizations/org-search.php
+    MARC_ORGANIZATION_CODE = "marc_organization_code"
+
+    INCLUDE_SUMMARY = u'include_summary'
+    INCLUDE_SIMPLIFIED_GENRES = u'include_simplified_genres'
+    STORAGE_PROTOCOL = u'storage_protocol'
+
+    LIBRARY_SETTINGS = [
+        { "key": MARC_ORGANIZATION_CODE,
+          "label": _("The MARC organization code for this library (003 field)."),
+          "description": _("MARC organization codes are assigned by the Library of Congress."),
+        },
+        { "key": INCLUDE_SUMMARY,
+          "label": _("Include summaries in MARC records (520 field)"),
+          "type": "select",
+          "options": [
+              { "key": "false", "label": _("Do not include summaries") },
+              { "key": "true", "label": _("Include summaries") },
+          ],
+          "default": "false",
+        },
+        { "key": INCLUDE_SIMPLIFIED_GENRES,
+          "label": _("Include Library Simplified genres in MARC records (650 fields)"),
+          "type": "select",
+          "options": [
+              { "key": "false", "label": _("Do not include Library Simplified genres") },
+              { "key": "true", "label": _("Include Library Simplified genres") },
+          ],
+          "default": "false",
+        },
+    ]
+
+    SETTINGS = [
+        { "key": STORAGE_PROTOCOL,
+          "label": _("Storage service protocol"),
+          "description": _("Storage protocol to use for uploading generated MARC files. The service must already be configured under 'Storage Services'."),
+          "type": "select",
+          "options": [{ "key": ExternalIntegration.S3, "label": ExternalIntegration.S3 }],
+          "default": ExternalIntegration.S3,
+        },
+    ]
```

Comment:
Rather than naming a storage protocol I think this should name a particular storage integration. Right now there can only be one storage integration sitewide, and when we allow more than one storage integration it's pretty likely we'll allow multiple integrations with the same protocol (because I think we'd change the 'S3' integration to refer to a specific S3 bucket rather than an overall set of credentials).

So for now, it would work to have this be a yes-or-no setting: upload it to storage or don't. But I do would like a way to say "this is a drop-down that lets you choose any configured setting with a certain goal."

Reply:
That would be better, but how would you represent it in the database?